consumption.tex

```diff
@@ -0,0 +1,87 @@
+
+A user who wishes to acquire a target cryptocurrency first has to form burn address which encodes her receiving address on the target blockchain. The user then sends an amount of the source cryptocurrency to that address. She then proceeds to submit a proof of this burn on the target blockchain, where the burn is verified and the user is credited an equivalent amount of target cryptocurrency on her receiving address.
+
+In accordance to the terminology laid out in~\cite{pow-sidechains}, we call this party the \emph{prover} and we call the smart contract the \emph{verifier}.
+
+We abstract the functionalities regarding transactions on the source blockchain into a \emph{blockchain transaction protocol}.
+
+\begin{definition}[Blockchain transaction protocol]\label{def:blockchain-tx}
+  A \emph{blockchain transaction protocol} $\Pi_\zeta$ consists of two functions $\VerTxIncl$ and $\TxHasTransfer$.
+
+  \begin{itemize}
+    \item $\VerTxIncl(b, tx, \tau)$: Given block $b$, transaction $tx$ and a proof $\tau$, return $\true$ if $\tau$ is a valid proof for the fact that $tx \in b$.
+    \item $\TxHasTransfer(tx, addr, amount)$: Given transaction $tx$, an address $addr$ and an amount, return $\true$ if $tx$ contains a value transfer of the specified amount towards the specified address.
+  \end{itemize}
+\end{definition}
+
+More concretely, in Bitcoin, each block header contains a commitment to the set of transaction ids in the block in the form of a Merkle Tree root. Thus, for Bitcoin the proof $\tau$ for $\VerTxIncl$ is a Merkle Tree inclusion proof. Ethereum stores a similar commitment in its header, in the form of the root of a Merkle-Patricia Trie~\cite{wood2014ethereum}.
+
+For a verifier to be convinced that a burn has taken place on a source blockchain, they need to ensure that the burn is contained in a transaction which belongs to the source blockchain. Specifically, the following data need to be supplied to the smart contract:
+
+\begin{itemize}
+  \item \textsf{tx}: The transaction which contains the burn on the source blockchain.
+  \item \textsf{blk}: The block where the aforementioned transaction is contained.
+  \item \textsf{txInclusion}: An inclusion proof showing that $\mathsf{tx} \in \mathsf{blk}$ such that $\VerTxIncl(\textsf{blk}, \textsf{tx}, \textsf{txInclusion}) = \true$.
+  \item \textsf{blockConnection}: A proof that \textsf{blk} is contained in the best source blockchain.
+\end{itemize}
+
+Putting these together, the final proof is the tuple $\beta = \sf(tx, txInclusion, blk, blockConnection)$.
+
+\subsection{Proof-of-Work Sidechains}
```

Comment:
this subsection is potentially unnecessary

Reply:
I agree. The detailed description of the crosschain smart contract is unnecessary. The presentation of the three functions of the crosschain smart contact is also unnecessary. We can just concisely state that there are two mechanisms to check whether a block belongs to the best remote blockchain without having the target blockchain miners connect to it or download it: (1) Through a federation or (2) through a decentralized proof. Then just use `is-in-best-chain(b)`. Then give a few words about how the federation works (as you do) and how the decentralized proof works (perhaps a paragraph talking about the fact that anyone can submit a small proof called a NIPoPoW which is a short string that compresses the proof-of-work of the remote chain and this is sufficient for verification). A citation should then suffice for the interested reader.